tests/test_shuffle.py

```diff
@@ -1,12 +1,70 @@
+import uuid
+
 import dask
+import dask.dataframe as dd
+import pytest
+
+
+@pytest.fixture(scope="session")
+def s3_stability_url(s3, s3_bucket):
+    # Ensure that the stability-scratch directory exists,
+    # but do NOT reomove it as multiple test runs could be
+    # accessing it at the same time
+    stability_url = f"{s3_bucket}/stability-scratch"
+    s3.mkdirs(stability_url, exist_ok=True)
+    return stability_url
+
+
+@pytest.fixture(scope="session")
+def s3_test_data_url(s3, s3_stability_url):
+    # Unique, because multiple tests are accessing the bucket
+    test_url = f"{s3_stability_url}/test-data-{uuid.uuid4().hex}"
+
+    try:
+        s3.makedirs(test_url)
+        yield test_url
+    finally:
+        s3.rm(test_url, recursive=True)
+
 
+@pytest.fixture
+def s3_write_url(s3, s3_stability_url):
```

Comment:
Nitpick: I think we can drop `write` and still maintain the existing meaning. It would be good for the fixture tests use to say "give me an s3 bucket I can use" to have concise (but still meaningful) name

```suggestion
def s3_url(s3, s3_stability_url):
```

Reply:
- I've created a `s3_url_factory` fixture in `conftest.py`.
- It's a factory fixture because with the removal of the `shuffle_dataset` fixture, multiple scratch directories are necessary in `test_shuffle` to:

  1. Generate the test dataset
  2. Write the shuffled dataset

- It's not possible to use a standard fixture twice in a test